File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/UpdateUserUseCase.java
package br.org.restaurantepopular.usecase.user;

import br.org.restaurantepopular.entity.User;
import br.org.restaurantepopular.usecase.core.IUseCase;

import java.util.Optional;

public class UpdateUserUseCase implements IUseCase<User, Optional> {

    private final IUserRepository userRepository;

    public UpdateUserUseCase(IUserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public Optional execute(User user) {

        userRepository.update(user);
        return Optional.empty();
    }
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/user/UpdateUserUseCase.kt
package br.org.restaurantepopular.usecase.user

import br.org.restaurantepopular.entity.User
import br.org.restaurantepopular.usecase.core.IUseCase

class UpdateUserUseCase(
    private val userRepository: IUserRepository
) : IUseCase<User, Unit> {

    override fun execute(request: User){
        userRepository.update(request)
    }
}